openrr-planner/examples/reach.rs

```diff
@@ -98,11 +105,24 @@ impl CollisionAvoidApp {
             ignore_rotation_z,
             self_collision_pairs,
             urdf_robot,
+            reference_robot,
         }
     }
 
     fn update_robot(&mut self) {
-        // this is hack to handle invalid mimic joints
+        // update the reference robot
+        self.reference_robot.set_joint_positions_clamped(
+            &self
+                .planner
+                .path_planner
+                .robot_collision_detector
+                .robot
+                .joint_positions(),
```

Comment:
I'm not sure about this, I feel it's better to define some functions to access `planner.path_planner.robot_collision_detector.robot` to capsulize the objects' members. (And make some fields private in the near future.)
For example, `planner.planned_joint_positions() = planner.path_planner.robot_collision_detector.robot.joint_position();`,  if this is not a special case.

What do you think?

Reply:
I agree with your comment, and I have also planned to capsulize so.
However, making robot_collision_detector in JointPathPlanner private requires a little large change. I would like to tackle this issue in another PR. Thank you!